conans/client/graph/range_resolver.py

```diff
@@ -78,14 +78,26 @@ def _search_remote_recipes(self, remote, search_ref):
             pattern_cached.update({remote.name: results})
         return results
 
+    @staticmethod
+    def _should_update(search_ref, update):
+        if update is None:
+            return False
+        assert isinstance(update, list)
+        # Legacy syntax had --update without pattern, that manifests as a True in the list
+        # (Usually one element, but support having more than one just in case)
+        # Note that it can't be false, the True is a flag set by Conan to update all
+        if any(isinstance(u, bool) for u in update):
+            return True
+        return any(search_ref.matches(pattern, is_consumer=False) for pattern in update)
+
     def _resolve_remote(self, search_ref, version_range, remotes, update):
         update_candidates = []
         for remote in remotes:
             remote_results = self._search_remote_recipes(remote, search_ref)
             resolved_version = self._resolve_version(version_range, remote_results,
                                                      self._resolve_prereleases)
             if resolved_version:
-                if not update:
+                if not self._should_update(search_ref, update):
```

Comment:
There are a few places still that this pattern should be applied to (`graph_binaries.py` compatibles handling for example), this was the minimal change to get the tests working